cli/central-client/src/main/java/org/ballerinalang/central/client/model/PackageResolutionRequest.java

```diff
@@ -129,7 +131,8 @@ public PackageResolutionRequest() {
     }
 
     public void addPackage(String orgName, String name, String version, Mode mode) {
-        packages.add(new Package(orgName, name, version, mode));
+        // The version is encoded to avoid issue handling the dash in pre-release version tag
+        packages.add(new Package(orgName, name, URLEncoder.encode(version, StandardCharsets.UTF_8), mode));
```

Comment:
Shall we encode the full JSON string in UTF-8 at the point of adding to the ok http.
```
new String(bytes, StandardCharsets.UTF_8);
```

Reply:
@hevayo 
At https://github.com/ballerina-platform/ballerina-lang/blob/master/cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java#L565 there's no bytes involved.

What about doing the following?
\-            RequestBody requestBody = RequestBody.create(JSON, new Gson().toJson(request));
\+            RequestBody requestBody = RequestBody.create(JSON, new String(new Gson().toJson(request).getBytes(), StandardCharsets.UTF_8));